### DISCUSSION TITLE
Check your files for NWB Best Practices

### DISCUSSION THREAD

**Discussion by CynthiaEzeh** [dandi/helpdesk#94] [Q&A]
Hi,

I am trying to upload a nwb file to this dataset:
after running this code: "nwbinspector <source_folder> --config dandi" in my newley created virtual environment, 

I got an error message which is:
Access is denied.

Please what could be wrong with the dataset?

Thanks!

**Comment by bendichter** [comment]
Can you please copy/paste the entire traceback?

**Reply by CynthiaEzeh** [reply]
base) C:\Users\ezehc>C:\Users\ezehc\anaconda3\python.exe -m pip install --upgrade pip

(base) C:\Users\ezehc>\Users\ezehc\anaconda3\python.exe -m pip --version

(base) C:\Users\ezehc>\Users\ezehc\anaconda3\python.exe -m pip install --user virtualenv

(base) C:\Users\ezehc>.\env\Scripts\activate

(env) (base) C:\Users\ezehc>where python
C:\Users\ezehc\env\Scripts\python.exe
C:\Users\ezehc\anaconda3\python.exe
C:\Users\ezehc\AppData\Local\Microsoft\WindowsApps\python.exe

(env) (base) C:\Users\ezehc>//github.com/GoogleCloudPlatform/google-auth-library-python.git
The network path was not found.

(env) (base) C:\Users\ezehc>/github.com/GoogleCloudPlatform/google-auth-library-python.git
The system cannot find the path specified.

(env) (base) C:\Users\ezehc>pip install -U dandi
Collecting dandi
  Downloading dandi-0.53.0-py3-none-any.whl (232 kB)

(env) (base) C:\Users\ezehc>pip install -U nwbinspector

(env) (base) C:\Users\ezehc>nwbinspector <source_folder> --config dandi
The system cannot find the file specified.

(env) (base) C:\Users\ezehc>nwbinspector <source_folder> --config dandi
Access is denied.

**Reply by bendichter** [reply]
@CodyCBakerPhD do you have any idea why this might be happening? Is nwbinspector writing to disk?

**Reply by CodyCBakerPhD** [reply]
@CynthiaEzeh What is the name of the folder you want to run the Inspector on?

Do you know where it is located on your  system?

Do you have permission to access (read) from files in that location?

**Reply by CynthiaEzeh** [reply]
@CodyCBakerPhD  The folder name is source_folder. I followed the instruction on the Dandi website: https://www.dandiarchive.org/handbook/13_upload/, which says: Convert your data to NWB 2.1+ in a local folder. Let's call this <source_folder>. 

I created the source_folder and saved the nwb_file in the folder, but when I ran the inspector code, it said access is denied. The source folder is located C:\Users\ezehc\env\source_folder in my personal computer. 

Yes, I have permission as the admin of my personal computer. At first, I thought it was my company's desktop because I had no permission as the admin. But I repeated the same thing on my personal computer and got the same message.

Is there any specific folder that is required?

Thanks!

**Reply by CodyCBakerPhD** [reply]
> C:\Users\ezehc\env\source_folder

In that case, the command to run is `nwbinspector C:\Users\ezehc\env\source_folder`.

If you continue to experience issues, I'd be happy to meet in person to go through the process step-by-step! Just pick a time on [my calendly](https://calendly.com/cody-c-baker/) and it will setup a Zoom link for us.

**Reply by CynthiaEzeh** [reply]
Hi, @CodyCBakerPhD.

Thank you so much!
The code worked, but I scheduled a meeting with you to make sure that am doing things correctly since it's my first time uploading a Dandi set.
